### PR TITLE
Code copy button second script

### PR DESCRIPTION
**Pull request by ntorbinskiy**

### CODE REVIEW

File: components/Post/PostContent.tsx [hunk 1]
```diff
@@ -2,20 +2,28 @@ import config from "config";
...
-  return str.slice(openedElement, closedElement);
+  return string.slice(openedElement, closedElement);
 };
 
 const PostContent = ({ post }: Props) => {
   const description = getFirstParagraph(post.content);
+  const document = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    return renderCopyButtons(document);
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:25]: @ntorbinskiy I'm not sure you need a return statement in here, do you?

File: components/Post/PostContent.tsx [hunk 2]
```diff
@@ -2,20 +2,28 @@ import config from "config";
...
-const getFirstParagraph = (str: string) => {
-  const openedElement = str.indexOf("<p>") + 3;
-  const closedElement = str.indexOf("</p>");
+const getFirstParagraph = (string: string) => {
+  const openedElement = string.indexOf("<p>") + 3;
+  const closedElement = string.indexOf("</p>");
 
-  return str.slice(openedElement, closedElement);
+  return string.slice(openedElement, closedElement);
 };
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:18]: @ntorbinskiy Like the full names 👍

File: components/Post/PostContent.tsx [hunk 3]
```diff
@@ -2,20 +2,28 @@ import config from "config";
...
 
 const PostContent = ({ post }: Props) => {
   const description = getFirstParagraph(post.content);
+  const document = useRef<HTMLDivElement>(null);
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:22]: Like it in here, as well!

File: helpers/renderCopyButtons.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  buttonText.classList.add("btn-copy-text");
+  buttonText.innerHTML = "Copy";
+
+  button.appendChild(buttonIcon);
+  button.append(buttonText);
+
+  return button;
+};
+
+const renderCopyButtons = (doc: RefObject<HTMLDivElement>): (() => void) => {
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:105]: @ntorbinskiy rename it **document** just to be consistent

File: helpers/renderCopyButtons.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    });
+
+    item.addEventListener("mouseleave", () => {
+      buttonStyle.display = "none";
+    });
+    button.addEventListener("click", (event) => {
+      onClickCheck(event);
+    });
+  });
+  return (): void => {
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:134]: Add a break line before a return statement

File: components/Post/PostContent.tsx [hunk 6]
```diff
@@ -50,8 +58,7 @@ const PostContent = ({ post }: Props) => {
           </a>
         ))}
       </div>
-
-      <div dangerouslySetInnerHTML={{ __html: post.content }} />
+      <div ref={document} dangerouslySetInnerHTML={{ __html: post.content }} />
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:61]: Love it, that we don't have to attach another handler in here

File: helpers/renderCopyButtons.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  button.append(buttonText);
+
+  return button;
+};
+
+const renderCopyButtons = (doc: RefObject<HTMLDivElement>): (() => void) => {
+  const codeSnippets =
+    doc.current?.querySelectorAll("div.remark-highlight") ?? [];
+
+  codeSnippets.forEach((item) => {
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:109]: Rename **item** to **codeSnippet** name

File: helpers/renderCopyButtons.ts [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    }
+    item.prepend(createCopyButton());
+
+    const button = item.childNodes[0];
+
+    if (!(button instanceof HTMLElement)) {
+      return;
+    }
+
+    const buttonStyle = button.style;
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:121]: I'm not sure you need a separate variable for this button style's inner property styles, it just confuses
```suggestion
```

File: helpers/renderCopyButtons.ts [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+    item.addEventListener("mouseenter", () => {
+      buttonStyle.display = "flex";
+    });
+
+    item.addEventListener("mouseleave", () => {
+      buttonStyle.display = "none";
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:128]:
```suggestion
      button.style.display = "none";
```

File: components/Post/PostContent.tsx (hunk 1 above)
Comment (ntorbinskiy) [reply components/Post/PostContent.tsx:25]: @1alexvash 
No, I need it, because I return a teardown method that'll remove all EventListeners

Comment (1alexvash) [reply components/Post/PostContent.tsx:25]: @ntorbinskiy usually the cleanup looks like this:

```ts
  useEffect(() => {
    attachEventListeners();
    return () => removeEvenetListeners()
  }, [post.content])
```

🤔

File: helpers/renderCopyButtons.ts [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+  const areChildNodesAndClassesValid = (
+    buttonImg: ChildNode,
+    buttonText: ChildNode,
+    target: Target
+  ): buttonImg is HTMLElement => {
+    if (
+      buttonImg instanceof HTMLElement &&
+      buttonText &&
+      targetClasslistCheck(target)
```
Comment (1alexvash) [inline helpers/renderCopyButtons.ts:23]:
```suggestion
  buttonCopyClasses.includes(target.className)
```

And I think we can remove this method `targetClasslistCheck`

Only keep `buttonCopyClasses`
@ntorbinskiy

File: helpers/renderCopyButtons.ts (hunk 7 above)
Comment (1alexvash) [reply helpers/renderCopyButtons.ts:109]: Still not done

Comment (1alexvash) [reply helpers/renderCopyButtons.ts:109]: You've done it below but, not here

**Pull request merged**